src/sw.js

```diff
@@ -59,6 +68,35 @@ const SWTasks = {
   },
 };
 
+self.addEventListener("message", async(function*({data, source}) {
+  var result;
+  var {name, id} = data;
+  switch (name) {
+  case "NewTab:InitProviderManager":
+    yield ProviderManager.init();
+    result = true;
+    break;
+  case "NewTab:GetLinks":
+    result = yield Links.getLinks();
+    break;
+  case "NewTab:PopulateCache":
+    result = yield Links.populateCache();
+    break;
+  case "NewTab:CacheHistoryLinks":
+    PlacesProvider.setLinks(data.placesLinks);
+    result = true;
+    break;
+  default:
+    console.warn("Unhandled message", data.name);
+  }
+  // Only respond to messages that have an id, even if unhandeled.
+  // This is to prevents locking any expecting promises.
+  if (!id) {
```

Comment:
I think I made a boo boo here in my original code. I think we always want to respond or we will block promises. I suggest removing the if statement here and always responding.